### PR TITLE
Fixed issue #18209: Missing RemoteControl function to get the field mapping

### PR DESCRIPTION
**Pull request by gabrieljenik**
- Added get_fieldmap Remote Control function

### CODE REVIEW

File: application/helpers/remotecontrol/remotecontrol_handle.php [hunk 1]
```diff
@@ -3320,6 +3320,36 @@ public function get_uploaded_files($sSessionKey, $iSurveyID, $sToken)
...
     }
 
+    /**
+     * Get survey fieldmap (RPC function)
+     *
+     * Returns the requested survey's fieldmap in an array
+     *
+     * @access public
+     * @param string $sSessionKey Auth credentials
+     * @param int $surveyId ID of the Survey
```
Comment (Shnoulle) [inline application/helpers/remotecontrol/remotecontrol_handle.php:3330]: Optionnal param lang ?

**Review by Shnoulle — COMMENTED** [review]
See comments about language.

**Review by Shnoulle — APPROVED** [review]

**Review by c-schmitz — CHANGES_REQUESTED** [review]
Code looks fine, but please still add a test

File: tests/unit/helpers/RemoteControlTest.php [hunk 2]
```diff
@@ -535,4 +535,58 @@ public function testSetGroupProperties()
...
+        $survey = \Survey::model()->findByPk(self::$surveyId);
+        $group = $survey->groups[0];
+        $question = $group->questions[0];
+        $sgq = self::$surveyId . "X" . $group->gid . "X" . $question->qid;
+
+        // Check base language fieldmap
+        $result = $handler->get_fieldmap($sessionKey, self::$surveyId);
+        $this->assertEquals('Example question', $result[$sgq]['question']);
+
+        $result = $handler->get_fieldmap($sessionKey, self::$surveyId, 'es');
```
Comment (Shnoulle) [inline tests/unit/helpers/RemoteControlTest.php:589]: :+1: multilingual test !

**Review by c-schmitz — APPROVED** [review]

**Comment by c-schmitz** [comment]
Thanks. I also need this for 3.x

**Pull request merged**